src/BenchmarkDotNet/Columns/MetricColumn.cs

```diff
@@ -23,13 +23,28 @@ public class MetricColumn : IColumn
 
         public bool IsDefault(Summary summary, BenchmarkCase benchmarkCase) => false;
 
-        public bool IsAvailable(Summary summary) => summary.Reports.Any(report => report.Metrics.ContainsKey(descriptor.Id));
+        public bool IsAvailable(Summary summary) => descriptor is Diagnosers.AllocatedMemoryMetricDescriptor || summary.Reports.Any(IsAvailable);
+
+        private bool IsAvailable(BenchmarkReport report)
+        {
+            if (!report.Metrics.TryGetValue(descriptor.Id, out var metric))
+            {
+                return false;
+            }
+            if (metric.Descriptor is Diagnosers.MemoryDiagnoser.GarbageCollectionsMetricDescriptor)
```

Comment:
I don't like these explicit dependencies on specific diagnosers in the base `MetricColumn` class. Please, introduce a corresponding property in `IMetricDescriptor` that controls this behavior.

Reply:
Yeah, I thought that would be cleaner, but I didn't want to break the interface since it's public (and netstandard2.0 does not support DIM). I refactored it to work in a more abstract way without changing the interfaces, but if you don't mind breaking changes, I can update the interfaces instead.

Reply:
@timcassell the new approach with a hidden *static* state looks too hacky and may lead to unpleasant side effects. It's OK to extend the interface.

Reply:
Ok, I updated the interfaces. Since that's a breaking change, I think this should go into 0.14.0 instead of 0.13.4. What do you think?